### PR TITLE
Logit Boost

### PR DESCRIPTION
**Pull request by andrewdalpino**
Logit Boost is a stage-wise additive ensemble that uses regression trees to iteratively learn a logistic regression model for binary classification problems.

See

- https://web.stanford.edu/~hastie/Papers/AdditiveLogisticRegression/alr.pdf
- J. H. Friedman. (2001). Greedy Function Approximation: A Gradient Boosting Machine.
- J. H. Friedman. (1999). Stochastic Gradient Boosting.
- Y. Wei. et al. (2017). Early stopping for kernel boosting algorithms: A general analysis with localized complexities.

### CODE REVIEW

File: src/Classifiers/LogitBoost.php [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+     */
+    protected \Rubix\ML\CrossValidation\Metrics\Metric $metric;
+
+    /**
+     * The ensemble of weak regressors.
+     *
+     * @var mixed[]
+     */
+    protected array $ensemble = [
+        //
```
Comment (torchello) [inline src/Classifiers/LogitBoost.php]: Is this supposed to be empty?

File: src/Classifiers/LogitBoost.php [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+        ?Metric $metric = null
+    ) {
+        if ($booster and !in_array(get_class($booster), self::COMPATIBLE_BOOSTERS)) {
+            throw new InvalidArgumentException('Booster is not compatible'
+                . ' with the ensemble.');
+        }
+
+        if ($rate <= 0.0 or $rate > 1.0) {
+            throw new InvalidArgumentException('Learning rate must be'
+                . " greater than 0, $rate given.");
```
Comment (torchello) [inline src/Classifiers/LogitBoost.php]:
```suggestion
                . " between 0 and 1, $rate given.");
```

File: src/Classifiers/LogitBoost.php [hunk 3]
```diff
@@ -0,0 +1,629 @@
...
+    }
+
+    /**
+     * Return the string representation of the object.
+     *
+     * @return string
+     */
+    public function __toString() : string
+    {
+        return 'Gradient Boost (' . Params::stringify($this->params()) . ')';
```
Comment (torchello) [inline src/Classifiers/LogitBoost.php]:
```suggestion
        return 'Logit Boost (' . Params::stringify($this->params()) . ')';
```

**Review by torchello — COMMENTED** [review]
Looks good in general. Left some minor comments.

File: src/Classifiers/LogitBoost.php (hunk 1 above)
Comment (andrewdalpino) [reply src/Classifiers/LogitBoost.php]: It doesn't need to be, nice catch!

File: src/Classifiers/LogitBoost.php (hunk 3 above)
Comment (andrewdalpino) [reply src/Classifiers/LogitBoost.php]: Nice catch

**Pull request closed without merging**